docs/source/design.rst

```diff
@@ -1,7 +1,201 @@
-.. _internals:
+.. _design:
 
-*********************
-Ibis design internals
-*********************
+Design
+======
 
-More to come here.
+
+.. _primary_goals:
+
+Primary Goals
+-------------
+
+#. Type safety
+#. Expressiveness
+#. Composability
+#. Familiarty
+
+.. _flow_of_execution:
+
+Flow of Execution
+-----------------
+
+#. User writes expression
+#. Each method or function call builds a new expression
+#. Expressions are type checked as you create them
+#. Some expressions have some optimizations that happen as the user builds them
+   (EXAMPLE)
+#. Some optimizations happen at compile time? (EXAMPLE)
+#. Expressions are compiled
+#. The SQL string that generated by the compiler is sent to the database and
+   executed (this step is skipped for the Pandas backend)
+#. The database returns some data that is then turned into a pandas DataFrame
+   by ibis
```

Comment:
We should decide on `Ibis` vs `ibis` in prose and be consistent (pandas is of course styled as lowercase)

Reply:
I'll arbitrarily decide to follow the `pandas` convention.